projects/wave-core-new/src/lib/layer-collections/layer-collection-dropdown/layer-collection-dropdown.component.ts

```diff
@@ -0,0 +1,140 @@
+import {Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef, Input, EventEmitter, Output} from '@angular/core';
+import {
+    LayerCollectionDict,
+    LayerCollectionItemDict,
+    ProviderLayerCollectionIdDict,
+    ProviderLayerIdDict,
+} from '../../backend/backend.model';
+import {LayerCollectionService} from '../layer-collection.service';
+
+@Component({
+    selector: 'wave-layer-collection-dropdown',
+    templateUrl: './layer-collection-dropdown.component.html',
+    styleUrls: ['./layer-collection-dropdown.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LayerCollectionDropdownComponent implements OnInit {
+    @Input() root?: ProviderLayerCollectionIdDict = undefined;
+    @Input() rootLabel = 'Layers';
+    @Input() preselectedPath: Array<string | number> = []; // preselect entries in hierarchy either by name or index
+
+    @Output() layerSelected = new EventEmitter<ProviderLayerIdDict>();
+
+    items: Array<Array<LayerCollectionItemDict>> = [];
+    labels: Array<string> = [];
+    descriptions: Array<string> = [];
+    properties: Array<Array<[string, string]>> = [];
+
+    selections: Array<LayerCollectionItemDict> = [];
```

Comment:
and them (readonly) as well, right?

Reply:
705855365489b97a3bb92962d48357e7e6cca961